src/io/react-native/react-native-worker.ts

```diff
@@ -223,16 +225,15 @@ const fetchCorsProxy = async (
   uri: string,
   opts: EdgeFetchOptions = {}
 ): Promise<Response> => {
-  const randomIndex = Math.min(
-    PROXY_SERVER_URLS.length - 1,
-    Math.floor(Math.random() * PROXY_SERVER_URLS.length)
+  const shuffledUrls = shuffle([...PROXY_SERVER_URLS])
+  const tasks = shuffledUrls.map(proxyServerUrl => async () =>
+    await window.fetch(proxyServerUrl, {
+      headers: {
+        ...opts.headers,
+        'x-proxy-url': uri
+      },
+      ...opts
```

Comment:
oops. Didn't catch this in the original review, but `...opts` here would overwrite `headers` if `opts.headers` existed

```
> opts = { headers: {h1: 1, h2: 2}, opts2: 'opt2' }
{ headers: { h1: 1, h2: 2 }, opts2: 'opt2' }
> fetchOpts = { headers: {...opts.headers}}
{ headers: { h1: 1, h2: 2 } }
> 
> fetchOpts = { headers: {...opts.headers, proxy: 'myuri' }}
{ headers: { h1: 1, h2: 2, proxy: 'myuri' } }
> 
> fetchOpts = { headers: {...opts.headers, proxy: 'myuri' }, ...opts}
{ headers: { h1: 1, h2: 2 }, opts2: 'opt2' }
> 
```

Should put the `...opts` before the `headers`

Reply:
That was a dumb mistake on my part. Good catch.